content/master/operations/operation.md

```diff
@@ -0,0 +1,580 @@
+
+### Events
+
+Operations emit Kubernetes events for important activities:
+- Function run results and warnings
+- Resource apply failures
+- Operation lifecycle events (creation, completion, failure)
+
+### Troubleshooting operations
+
+**Select operation status:**
+
+```shell
+kubectl get operation my-operation -o wide
+kubectl describe operation my-operation
+```
+
+**Common failure scenarios:**
+
+1. **ValidPipeline condition is False** - Function doesn't support operations:
+   ```yaml
+   conditions:
+   - type: ValidPipeline
+     status: "False"
+     reason: InvalidFunctionCapability
+     message: "Function function-name doesn't support operations"
+   ```
+   *Solution*: use a function that declares `operation` capability.
+
+2. **Succeeded condition is False** - Function run failed:
+   ```yaml
+   conditions:
+   - type: Succeeded
+     status: "False" 
+     reason: PipelineFailure
+     message: "Function returned error: connection timeout"
+   ```
+   *Solution*: view function logs and fix the underlying issue.
+
+3. **Resource apply failures** - View events for details:
+   ```shell
+   kubectl get events --field-selector involvedObject.name=my-operation
+   ```
+
+**Debug function runs:**
+
+```shell
+# View function logs
+kubectl logs -n crossplane-system deployment/function-python
+
+# Check operation events
+kubectl get events --field-selector involvedObject.kind=Operation
+
+# Inspect operation status in detail
+kubectl get operation my-operation -o jsonpath='{.status.pipeline}' | jq '.'
+```
+
+## Resource management
+
+Operations can create or change any Kubernetes resources using server-side
+apply with force ownership. This means:
+
+**What Operations can do:**
+- Create new resources of any kind
+- Change existing resources by taking ownership of specific fields
+- Apply changes that may conflict with other controllers
+
+**What Operations can't do:**
+- Delete resources (current limitation of alpha implementation)
+- Establish owner references (resources aren't garbage collected)
+- Continuously maintain desired state (they run once)
+
+{{<hint "important">}}
+Use caution with Operations that change resources managed by other controllers.
+Operations force ownership when applying changes, which can cause conflicts.
+{{</hint>}}
+
+## Test an operation
+
+You can preview the output of any Operation using the Crossplane CLI. You
+don't need a Crossplane control plane to do this. The Crossplane CLI uses Docker
+Engine to run functions.
+
+{{<hint "tip">}}
+See the [Crossplane CLI docs]({{<ref "../cli">}}) to
+learn how to install and use the Crossplane CLI.
+{{< /hint >}}
+
+{{<hint "important">}}
+Running `crossplane alpha render op` requires [Docker](https://www.docker.com).
+{{< /hint >}}
+
+Provide an operation, composition functions, and any required resources to render
+the output locally.
+
+```shell
+crossplane alpha render op operation.yaml functions.yaml --required-resources=ingress.yaml
```

Comment:
should we be providing these resources on this page for folks to be able to run this command? or is it more for demonstrative purposes and we don't expect the reader to run this?